gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/ProjectInfo.java

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.gcloud.resourcemanager;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.collect.ImmutableMap;
+
+import org.joda.time.DateTime;
+import org.joda.time.format.ISODateTimeFormat;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * A Google Cloud Resource Manager project metadata object.
+ */
+public class ProjectInfo implements Serializable {
+
+  private static final long serialVersionUID = 9148970963697734236L;
+  private final String name;
+  private final String id;
+  private final Map<String, String> labels;
+  private final Long number;
+  private final State state;
+  private final Long createTimeMillis;
+  private final ResourceId parent;
+
+  public enum State {
+    LIFECYCLE_STATE_UNSPECIFIED,
+    ACTIVE,
+    DELETE_REQUESTED,
+    DELETE_IN_PROGRESS;
+  }
+
+  public static class Builder {
+    private String name;
+    private String id;
+    private Map<String, String> labels;
+    private Long number;
+    private State state;
+    private Long createTimeMillis;
+    private ResourceId parent;
+
+    Builder() {
+      labels = new HashMap<String, String>();
+    }
+
+    public Builder name(String name) {
+      this.name = name;
+      return this;
+    }
+
+    public Builder id(String id) {
+      this.id = id;
+      return this;
+    }
+
+    public Builder addLabel(String key, String value) {
+      this.labels.put(key, value);
+      return this;
+    }
+
+    public Builder removeLabel(String key) {
+      this.labels.remove(key);
+      return this;
+    }
+
+    public Builder clearLabels() {
+      this.labels.clear();
+      return this;
+    }
+
+    public Builder labels(Map<String, String> labels) {
+      this.labels = checkNotNull(labels);
```

Comment:
We should create a new hashmap here, to avoid changes to occur out of the builder:
```java
this.labels = Maps.newHashMap(checkNotNull(labels));
```